app/dashboard/bookings/actions.ts

```diff
@@ -54,4 +54,12 @@ export async function deleteBooking(bookingId: string) {
   return prisma.booking.delete({
     where: { id: bookingId },
   });
-}   
\ No newline at end of file
+}
+
+export async function totalRevenue(): Promise<number> {
+  const bookings = await getAllBookings();
+  return bookings.reduce((sum, booking) => {
+    const revenue = Number(booking.revenue);
+    return sum + (isNaN(revenue) ? 0 : revenue);
+  }, 0);
```

Comment:
Keep for now, but keep in mind that eventually we'll want this to be a db level query. I believe that right now this would do a `select * from bookings` which is not scalable.